langserver/tracing.go

```diff
@@ -23,6 +23,12 @@ func (h *HandlerCommon) InitTracer(conn *jsonrpc2.Conn) {
 		return
 	}
 
+	if t := opentracing.GlobalTracer(); t != nil {
+		h.tracer = t
```

Comment:
Ugh, it seems very wrong that `HandlerCommon` has a `tracer` field. A tracer should always be a global singleton. Why is it built that way?